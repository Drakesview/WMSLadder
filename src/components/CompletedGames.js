import React from 'react';
import moment from 'moment';

export const CompletedGames = (props) => {
console.log(props)
return (
    <div>
        <div>
            {props.player1Name} {props.Player1Score}  
            {props.player1Name === props.WinningPlayer ? 
                <span> Winner</span> : <span> Loser</span>} 
        </div>
        vs
        <div>
            {props.player2Name} {props.Player2Score}
            {props.player2Name === props.WinningPlayer ? 
                <span> Winner</span> : <span> Loser</span>} 
        </div>
        <p>{moment(props.datePlayed).format('Do MMMM YYYY')}</p>
    </div>


)}

export default CompletedGames