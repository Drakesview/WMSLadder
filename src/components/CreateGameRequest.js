import React from 'react'
import {connect} from 'react-redux'
import moment from 'moment'
import {SingleDatePicker} from 'react-dates'
import {getLadder, startUpdateLadder} from '../actions/ladder'
import {startAddMatch} from '../actions/matches'
import uuid from 'uuid'

const now = moment()

export class ResultsPage extends React.Component{
    constructor(props) {
        super(props)
        this.state= {
            player1:this.props.players.find(obj => {
                return obj.id === this.props.auth.uid
            }).name,
            player2:'',
            player1Id:this.props.auth.uid,
            datePlayed: moment(),
            calendarFocused:false,
            error:''
        }
        this.handleChange = this.handleChange.bind(this)
    }
    handleChange(e) {
        this.setState({player2: e.target.value})
        // this.setState(() => ({player2:e.target.value}))
    }
    onDateChange = (datePlayed) => {
        if(datePlayed) {
            this.setState(() => ({datePlayed}))
        }
    }

    onFocusChange = ({ focused }) => {
        this.setState(() => ({calendarFocused:focused}))
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (this.state.player2 === '') {
             this.setState(() => ({error:'Please select an opponent'}))
        } else {
            this.props.startAddMatch({
                player1Name:this.state.player1,
                Player1:this.props.auth.uid,
                Player1Score:0,
                player2Name:this.state.player2,
                Player2:this.props.players.find(obj => {
                    return obj.name === this.state.player2
                }).id,
                Player2Score:0,
                WinningPlayer:'',
                LosingPlayer:'',
                datePlayed:this.state.datePlayed.valueOf(),
                stage:2
            })
            this.props.history.goBack()
        }
    }

    render() {
    return (
        <div>
        <form onSubmit = {this.onSubmit}>
            <div>
            {this.state.error && <p>{this.state.error}</p>}
                <h3>Player 1</h3>
                {this.state.player1}
            </div>
        <div>
            <h3>Player 2</h3>
            <select name={this.state.player2} onChange={this.handleChange}>
                <option value=''>Pick Opponent</option>
                {this.props.players.map(({name,id}) => {
                    if(name !== this.state.player1){
                    return <option 
                            key ={id}
                            value={name}        
                    >{name}</option>
                }})}
            </select>
            <SingleDatePicker
                 date={this.state.datePlayed}
                 onDateChange={this.onDateChange}
                 focused={this.state.calendarFocused}
                 onFocusChange={this.onFocusChange}
                 numberOfMonths={1}
                 isOutsideRange={() => false}
                />
        </div>
            <button>Send Game Request</button>
            </form>
        </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        players:state.ladder,
        auth:state.auth
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddMatch: (matchData) => dispatch(startAddMatch(matchData)),
})

export default connect(mapStateToProps,mapDispatchToProps)(ResultsPage)
