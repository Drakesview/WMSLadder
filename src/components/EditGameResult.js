import React from 'react'
import {connect} from 'react-redux'
import moment from 'moment'
import {SingleDatePicker} from 'react-dates'
import {startUpdateLadder} from '../actions/ladder'
import {recordGameScore} from '../actions/matches'

const now = moment()

export class EditGameResult extends React.Component{
    constructor(props) {
        super(props)
        this.state = {
            player1:props.matches ? props.matches.player1Name: '',
            player2:props.matches ? props.matches.player2Name: '',
            winningPos:0,
            losingPos:0,
            player1Score:0,
            player2Score:0,
            error:'',
            player1CurrentIndex:props.matches ? props.players.findIndex((player) => {
                return player.name === props.matches.player1Name                    
            }) : -1,
            player2CurrentIndex:props.matches ? props.players.findIndex((player) => {
                return player.name === props.matches.player2Name}) : -3,
            affectedPlayers:[]
        }
    }
    onPlayer1ScoreChange = (e) => {
        const score1 = e.target.value
            if (!score1 || /^\d?$/.test(score1)) {
                this.setState(() => ({player1Score:score1}))
            }
    }

    onPlayer2ScoreChange = (e) => {
        const score2 = e.target.value
            if (!score2 || /^\d?$/.test(score2)) {
                this.setState(() => ({player2Score:score2}))
            }
    }
    winningPlayer = (player1Score, player2Score) => {
        const propPlayers = this.props.players
        const statePlayers = this.state
        if (player1Score > player2Score) {
            return this.props.players.map((player) => {
                return player.name
            }).indexOf(this.state.player1)
        } else if (player1Score < player2Score) {
            return propPlayers.map((player) => {
                return player.name
            }).indexOf(statePlayers.player2)
        } else {
            return -1
        }
    }
    losingPlayer = (player1Score, player2Score) => {
        const propPlayers = this.props.players
        const statePlayers = this.state
        if (player1Score < player2Score) {
            return this.props.players.map((player) => {
                return player.name
            }).indexOf(this.state.player1)
        } else if (player1Score > player2Score) {
            return propPlayers.map((player) => {
                return player.name
            }).indexOf(statePlayers.player2)
        } else {
           return -1
        }
    }
    onSubmit = (e) => {
        e.preventDefault()
        const propPlayers = this.props.players
        const statePlayers = this.state
        if (this.state.player2 === '') {
             this.setState(() => ({error:'Please select an opponent'}))
        } else {
            const winningPlayerIndex = this.winningPlayer(statePlayers.player1Score,statePlayers.player2Score)
            const losingPlayerIndex = this.losingPlayer(statePlayers.player1Score,statePlayers.player2Score)
            const winner = this.props.players[winningPlayerIndex]
            const loser = this.props.players[losingPlayerIndex]
            const affectedPlayers=[]
            if(losingPlayerIndex !== -1 && winningPlayerIndex !== -1) {
                if (losingPlayerIndex > winningPlayerIndex) {
                    //update winning and losing players game
                    //update the props.match details with score
                        //update the stage

                    winner.gamesWon += 1
                    winner.gamesPlayed += 1
                    loser.gamesLost += 1
                    loser.gamesPlayed += 1

                    affectedPlayers.push(winner)
                    affectedPlayers.push(loser)

                } else {

                    //find all players inbetween the winner and loser as they would
                    //need a new pos to be pos + 1 
                    const middlePlayers = propPlayers.filter((player,index) => {
                        return index > losingPlayerIndex && index < winningPlayerIndex
                    })
                    const newPos =middlePlayers.map((player) => {
                        return {...player,
                                pos:player.pos + 1}
                    })

                    newPos.forEach((player) => {
                        affectedPlayers.push(player)
                    })
                    winner.pos = losingPlayerIndex + 1
                    winner.gamesWon += 1
                    winner.gamesPlayed += 1
                    loser.pos = losingPlayerIndex + 2
                    loser.gamesLost += 1
                    loser.gamesPlayed += 1

                    affectedPlayers.push(winner)
                    affectedPlayers.push(loser)
                    
                   

                }
                //Call a update for each item in the affected players array to update each player
                //call update for the match details.
                //this.props.history.push('/')

                affectedPlayers.forEach((player) => {
                    this.props.startUpdateLadder(player)
                })

                this.props.recordGameScore({
                    id:this.props.matches.id,
                    LosingPlayer:loser.name,
                    WinningPlayer:winner.name,
                    Player1Score: this.state.player1Score,
                    Player2Score:this.state.player2Score,

                })
                this.props.history.push('/')

            } else {
                this.setState(() => ({error:'Games can not be a draw as they are a best of 5 please try again'}))
            }
        }
    }
    goBack = () => {
        this.props.history.goBack()
    }

    componentDidMount () {
        if (!this.props.matches) {
            this.props.history.goBack()
        }
    }
    render() {
    return this.props.matches ?  (
        <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Record result</h1>
            </div>
        </div>
        <div className="content-container">
            <form onSubmit={this.onSubmit} className="form">
            {this.state.error && <p className="form__error">{this.state.error}</p>}
                <h3>{this.state.player1}</h3>
                <input
                autoFocus
                className="text-input"
                placeholder="Player 1 Score"
                value={this.state.player1Score}
                type="number"
                onChange={this.onPlayer1ScoreChange}
                ></input>
                <h3>{this.state.player2}</h3>
                <input
                className="text-input"
                placeholder="Player 2 score" 
                value={this.state.player2Score}
                type="number"
                onChange={this.onPlayer2ScoreChange}
                />
                <div>
            <button className="button-layout">Save Result</button>
            </div>
            </form>
            <button className="form button-layout button-layout--secondary" onClick={this.goBack} >Cancel</button>
        </div>
        </div>

        
        )
        : (
            
            <div>
            
            <p>No game found please go back to profile</p></div>
        )
    }
}

const mapStateToProps = (state,props) => {
    return {
        matches:state.matches.find((match) => match.id === props.match.params.id),
        players:state.ladder
    }
}

const mapDispatchToPropbs = (dispatch) => {
    return {
        recordGameScore: (matchData) => dispatch(recordGameScore(matchData)),
        startUpdateLadder: (player) => dispatch(startUpdateLadder(player))
    }
}


export default connect(mapStateToProps,mapDispatchToPropbs)(EditGameResult)
