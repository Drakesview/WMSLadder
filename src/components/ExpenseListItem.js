import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
       <Link to={'/edit/'+id}> <h3>{description}</h3> </Link>
        <p>
        {numeral(amount / 100).format('$0,0.00')}
         - 
        {moment(createdAt).format('Do MMMM YYYY')}</p>
    </div>
)


//load a local for numeral
numeral.register('locale', 'uk', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    currency: {
        symbol: 'Â£'
    }
});

numeral.locale('uk');

export default ExpenseListItem