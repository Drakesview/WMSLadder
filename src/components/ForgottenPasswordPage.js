import React from 'react';
import {connect} from 'react-redux';
import validator from 'validator';
import { startEmailReset } from '../actions/auth'

export class ForgottenPasswordPage extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email:'',
            error:''
        }
    }
    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({email}))
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (validator.isEmail(this.state.email)) {
            this.props.startEmailReset(this.state.email)
            this.props.history.push('/')
        }  else {
            this.setState(() => ({error:'Please enter a valid email address'}))
        }
    }
render () {
    return (
        <div>
            <form onSubmit={this.onSubmit}>
            {this.state.error && <p>{this.state.error}</p>}
                <label>Email address</label>
                <input 
                    type="text"
                    placeholder="Enter email address"
                    value={this.state.email}
                    onChange={this.onEmailChange}
                />
                <button>Send Email</button>
            </form>
            <button>back</button> 
        </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEmailReset: (email) => dispatch(startEmailReset(email)) 
    }
} 

export default connect(undefined,mapDispatchToProps)(ForgottenPasswordPage)