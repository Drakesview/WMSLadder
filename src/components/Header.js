import React from 'react';
import {Link} from 'react-router-dom'
import { connect } from 'react-redux';
import { startLogout, startSendEmailVerification } from '../actions/auth';


export class Header extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            error:''
        }
    }
    sendEmail = (e) => {
        e.preventDefault()
        this.props.startSendEmailVerification()
    }

    render(){
    return (
        <header className="header">
            <div className="content-container">
                <div className="header__content">
                <Link className="header__title" to={"/dashboard"} >
                    <h1>WMS Squash Laddder</h1>
                </Link>
                <Link className="header__title" to={"/profile/"+this.props.id}>
                    <h1>Profile</h1>
                </Link>
                <button className="button-layout button-layout--link" onClick={this.props.startLogout}>Logout</button> 
                </div>
                {this.state.error && <p className="header__title">{this.state.error}</p>}
            </div>
            <div className="header__verifyEmail">
                <div className="verifyEmail__content">
                {!this.props.emailVerified && <p>Please verify email if you need another email please
                    <span>
                        <button className="button-layout button-layout--link button-layout--verif-link " onClick={this.sendEmail}>click here</button>
                    </span>
                    </p>}
                </div>
            </div>
        </header>
    )};
}


const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    startSendEmailVerification: () => dispatch(startSendEmailVerification())
})
const mapStateToProps = (state) => ({
    id:state.auth.uid,
    emailVerified:state.auth.emailVerified
})

export default connect(mapStateToProps,mapDispatchToProps)(Header);