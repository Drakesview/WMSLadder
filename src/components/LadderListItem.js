import React from 'react';

const LadderListItem = ({ id , name , pos, index }) => (
    <div>
        <h3>{index+1} . {name}</h3>
    </div>
)

export default LadderListItem