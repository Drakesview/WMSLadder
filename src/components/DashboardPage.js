import React from 'react';
import {Link} from 'react-router-dom'
import LadderPage from './LadderPage'
import Header from './Header'
const DashboardPage = () => (
    <div>
        <LadderPage/>
    </div>
);

export default DashboardPage