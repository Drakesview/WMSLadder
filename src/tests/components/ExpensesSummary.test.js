import React from 'react';
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'


test('should render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1290}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={125353343}/>)
    expect(wrapper).toMatchSnapshot();
});