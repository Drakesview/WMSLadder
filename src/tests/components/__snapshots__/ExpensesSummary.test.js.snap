// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with expenses 1`] = `
<div>
  <p>
     Viewing 
    3
     expenses 
    totalling 
    $1,141.95
     
  </p>
</div>
`;

exports[`should render ExpensesSummary without expenses 1`] = `<div />`;
